### PR TITLE
docs(samples): adds samples from documentation with correct region tags

### LINKED ISSUE #369: Add samples from documentation
Samples exist in the Data Catalog documentation on cloud.google.com that are not present here. These samples should be added to improve their discoverability and ensure consistency across the library and the docs.

### PR DESCRIPTION
**Pull request by steffnay**
Adds samples from DevSite documentation. Region tags included in added/updates files are correct.

The following additions are needed before merging:
- Samples updated to sample rubric standards
- File names changed to match region tags
- Tests added

Requested changes are limited to samples in the scope of this PR.
Ideally, another PR will update content and metadata of the other existing samples.

- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #369 ☕️

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/datacatalog/Quickstart.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+public class CreateTags {
+
+  public static void createTags() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String tagTemplateId = "my_tag_template";
+    createTags(projectId, tagTemplateId);
+  }
+
+  public static void createTags(String projectId, String tagTemplateId) {
```
Comment (averikitsch) [inline samples/snippets/src/main/java/com/example/datacatalog/Quickstart.java:47]: nit: Listing the params here too would be nice.

**Review by averikitsch — COMMENTED** [review]
Added a few comments that should be applied across samples. Will you add tests to this PR?

**Comment by pmakani** [comment]
Ref PR #392

File: samples/snippets/src/main/java/com/example/datacatalog/CreateCustomEntry.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DataCatalogClient dataCatalogClient = DataCatalogClient.create()) {
+
+      // 1. Environment cleanup: delete pre-existing data.
+      // Delete any pre-existing Entry with the same name
+      // that will be used in step 3.
+      try {
+        String entryName = EntryName.of(projectId, location, entryGroupId, entryId).toString();
+        dataCatalogClient.deleteEntry(entryName);
+        System.out.printf("\nDeleted Entry: %s", entryName);
```
[the remark this answers is not in the document]
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/example/datacatalog/CreateCustomEntry.java]: Updated to use `println`

File: samples/snippets/src/main/java/com/example/datacatalog/CreateCustomEntry.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      // Delete any pre-existing Template with the same name
+      // that will be used in step 4.
+      try {
+        dataCatalogClient.deleteTagTemplate(
+            DeleteTagTemplateRequest.newBuilder()
+                .setName(tagTemplateName)
+                .setForce(true)
+                .build());
+        System.out.printf("\nDeleted template: %s", tagTemplateName);
+      } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/example/datacatalog/CreateCustomEntry.java]: updated to catch `ApiException`

**Comment by stephaniewang526** [comment]
Samples are added in https://github.com/googleapis/java-datacatalog/pull/392. Hence, closing this duplicate PR.

**Pull request closed without merging**